File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        int[] answer = {};
        List<Integer> ansList = new ArrayList<>(); 
		int order = 0;
		int cnt = 0;
		while(order != progresses.length) {
			for(int i = 0; i <progresses.length; i++) {
				progresses[i] += speeds[i];
			}
			
			if(progresses[order] >= 100) {
				while(true) {
					order++;
					cnt++;
					if(order == progresses.length || progresses[order] < 100) {
						ansList.add(cnt);
						cnt = 0;
						break;
					}
				}
			}			
		}
		answer = new int[ansList.size()];
		for(int i = 0; i < ansList.size(); i++) {
			answer[i] = ansList.get(i);
		}
        return answer;
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    var answer = [];
    let taskTimes = [];
    let stack = [];
    for (let i = 0; i < progresses.length; i++) taskTimes.push(Math.ceil((100 - progresses[i]) / speeds[i]));
    for (let taskTime of taskTimes){
        if (stack[0] < taskTime){
            answer.push(stack.length);
            stack = [taskTime];
        } else {
            stack.push(taskTime);
        }
    }
    if (stack.length) answer.push(stack.length);
    return answer;
}
